fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PruneOlapScanPartition.java

```diff
@@ -76,10 +81,18 @@ public Rule build() {
                     partitionSlots.add(partitionSlot);
                 }
             }
+            NereidsSortedPartitionsCacheManager sortedPartitionsCacheManager = Env.getCurrentEnv()
+                    .getSortedPartitionsCacheManager();
             List<Long> manuallySpecifiedPartitions = scan.getManuallySpecifiedPartitions();
-            Map<Long, PartitionItem> idToPartitions;
+            Map<Long, PartitionItem> idToPartitions = ImmutableMap.of();
+            SortedPartitionRanges<Long> sortedPartitionRanges = null;
             if (manuallySpecifiedPartitions.isEmpty()) {
-                idToPartitions = partitionInfo.getIdToItem(false);
+                Optional<SortedPartitionRanges<?>> sortedPartitionRangesOpt = sortedPartitionsCacheManager.get(table);
+                if (sortedPartitionRangesOpt.isPresent()) {
+                    sortedPartitionRanges = (SortedPartitionRanges<Long>) sortedPartitionRangesOpt.get();
+                } else {
+                    idToPartitions = partitionInfo.getIdToItem(false);
+                }
```

Comment:
when sortedPartitionRangesOpt.isPresent(), idToPartitions will be empty map, in PartitionPruner.prune, may lead to result wrong when BooleanLiteral.TRUE.equals(partitionPredicate) is true.